usersession/agent/session_agent.go

```diff
@@ -113,12 +183,39 @@ func (s *SessionAgent) Start() {
 		}
 		return nil
 	})
+	s.tomb.Go(s.exitOnIdle)
 	systemd.SdNotify("READY=1")
 }
 
-var (
-	shutdownTimeout = 5 * time.Second
-)
+func (s *SessionAgent) exitOnIdle() error {
+	timer := time.NewTimer(s.IdleTimeout)
+	idle := true
+Loop:
+	for {
+		select {
+		case <-s.tomb.Dying():
+			break Loop
+		case <-timer.C:
+			s.tomb.Kill(nil)
+			break Loop
+		case <-s.idle.change:
+			newIdle := s.idle.isIdle()
```

Comment:
Do we (unit) test this part of the code?

Reply:
I've updated the `TestExitOnIdle` test to verify that the timeout is correctly extended when new requests are made.